File: Lesson0/Benchmark/rand_mul.py
import numpy as np
import timeit
import sys

def rand_mul(size):
    a = np.random.rand(size, size)
    b = np.random.rand(size, size)
    return np.dot(a, b)

def bench_mul(start, stop, out=sys.stdout):
    for i in range(start, stop + 1):
        execution_times = timeit.repeat(lambda: rand_mul(i), repeat=10, number=8)
        min_execution_time = min(execution_times)
        print(f"{i}\t{min_execution_time}", file=out)

if __name__ == "__main__":
    with open("rand_mul.py.dat", "w") as file:
        bench_mul(1, 100, out=file)  # Example range for benchmarking

File: Lesson0/Benchmark/rand_mul.jl
using BenchmarkTools

function rand_mul(size::Int)
    a = rand(size, size)
    b = rand(size, size)
    return a * b
end

function bench_mul(start::Int, stop::Int; samples::Int=10_000, evals::Int=1, out::IOStream=stdout)
    for i in start:stop
        println(out, i, "\t", @belapsed rand_mul($i) samples = samples evals = evals)
    end
end

file = open("rand_mul.jl.dat", "w")
bench_mul(1, 100; samples=10, evals=8, out=file)
close(file)